ecommerce/frontend/src/components/UpdateProfileForm.js

```diff
@@ -0,0 +1,277 @@
+import { Form, Input, Button, Tooltip, InputNumber } from 'antd'
+import { QuestionCircleOutlined } from '@ant-design/icons'
+import { useState } from 'react'
+
+// Constants for layout
+const formItemLayout = {
+    labelCol: {
+        xs: {
+            span: 24,
+        },
+        sm: {
+            span: 8,
+        },
+    },
+    wrapperCol: {
+        xs: {
+            span: 24,
+        },
+        sm: {
+            span: 16,
+        },
+    },
+}
+const tailFormItemLayout = {
+    wrapperCol: {
+        xs: {
+            span: 24,
+            offset: 0,
+        },
+        sm: {
+            span: 16,
+            offset: 8,
+        },
+    },
+}
+
+export const UpdateProfileForm = ({ onSubmit = () => {} }) => {
+    const [form] = Form.useForm()
+
+    const initialValues = {
+        userType: 'customer',
+        name: 'Furkan',
+        surname: 'Varol',
+        username: 'vaveyla',
+        email: 'furkan.varol@gmail.com',
+        phone: {
+            countryCode: '+4',
+            number: '8452391058',
+        },
+    }
+
+    const onFinish = values => {
+        values.userType = initialValues.userType
+        if (isUpdateRequired(values)) {
+            console.log('Received values of form: ', values)
+            console.log('Initial Values: ', initialValues)
+            onSubmit(values)
+        } else {
+            console.log('Nothing is changed')
+        }
+    }
+
+    const onFinishFailed = errors => {
+        console.log('Finish failed: ', errors)
+    }
+
+    const isUpdateRequired = newValues => {
+        const keys = Object.keys(initialValues)
+
+        for (let key of keys) {
+            if (key === 'phone') {
+                if (
+                    initialValues[key]['countryCode'] != newValues[key]['countryCode'] ||
+                    initialValues[key]['number'] != newValues[key]['number']
+                ) {
+                    return true
+                }
+                continue
+            } else if (initialValues[key] !== newValues[key]) {
+                return true
+            }
+        }
+        return false
+    }
```

Comment:
A better more general way would be to use one of Ramda's functionsl like `R.equals`  https://ramdajs.com/docs/#equals

We are already using Ramda in another feature btw.